src/main/java/cloud/fogbow/ras/core/plugins/interoperability/aws/compute/v2/AwsV2ComputePlugin.java

```diff
@@ -1,47 +1,459 @@
+			}
+		}
+		throw new InstanceNotFoundException();
+	}
+	
+	protected List<InstanceNetworkInterfaceSpecification> loadNetworkInterfaces(ComputeOrder computeOrder) {
+		List<String> networkIds = getNetworkIds(computeOrder);
+		List<InstanceNetworkInterfaceSpecification> networkInterfaces = new ArrayList<>();
+		InstanceNetworkInterfaceSpecification nis = null;
+		for (String networkId : networkIds) {
+			nis = createInstanceNetworkInterface(networkId);
+			networkInterfaces.add(nis);
+		}
+		return networkInterfaces;
+	}
 
-    @Override
-    public boolean hasFailed(String instanceState) {
-        return true;
-    }
+	protected List<String> getNetworkIds(ComputeOrder computeOrder) {
+		List<String> networkIds = new ArrayList<String>();
+		networkIds.add(this.subnetId);
+		if (computeOrder.getNetworkIds() != null) {
+			networkIds.addAll(computeOrder.getNetworkIds());
+		}
+		return networkIds;
+	}
+
+	protected InstanceNetworkInterfaceSpecification createInstanceNetworkInterface(String networkId) {
+		InstanceNetworkInterfaceSpecification networkInterface = InstanceNetworkInterfaceSpecification.builder()
+				.subnetId(networkId) // Default sub-net in the available zone of the selected region.
+				.deviceIndex(FIRST_POSITION) // The first position of the network interface.
+				.groups(this.securityGroup).build();
+		
+		return networkInterface;
+	}
+
+	protected CreateTagsRequest createInstanceTagName(ComputeOrder computeOrder, String instanceId) {
+		String name = defineInstanceName(computeOrder.getName());
+
+		Tag tagName = Tag.builder()
+				.key(AWS_TAG_NAME)
+				.value(name)
+				.build();
+
+		CreateTagsRequest tagRequest = CreateTagsRequest.builder()
+				.resources(instanceId)
+				.tags(tagName)
+				.build();
+
+		return tagRequest;
+	}
+
+	protected String defineInstanceName(String instanceName) {
+		return instanceName == null ? SystemConstants.FOGBOW_INSTANCE_NAME_PREFIX + getRandomUUID() : instanceName;
+	}
+
+	// This method is used to aid in the tests
+	protected String getRandomUUID() {
+		return UUID.randomUUID().toString();
+	}
+
+	protected AwsHardwareRequirements findSmallestFlavor(ComputeOrder computeOrder, AwsV2User cloudUser)
+			throws InvalidParameterException, UnexpectedException, ConfigurationErrorException,
+			NoAvailableResourcesException {
+
+		AwsHardwareRequirements bestFlavor = getBestFlavor(computeOrder, cloudUser);
+		if (bestFlavor == null) {
+			throw new NoAvailableResourcesException(Messages.Exception.NO_MATCHING_FLAVOR);
+		}
+		return bestFlavor;
+	}
+
+	protected AwsHardwareRequirements getBestFlavor(ComputeOrder computeOrder, AwsV2User cloudUser)
+			throws InvalidParameterException, UnexpectedException, ConfigurationErrorException {
+
+		updateHardwareRequirements(cloudUser);
+		TreeSet<AwsHardwareRequirements> resultset = getFlavorsByRequirements(computeOrder.getRequirements());
+		for (AwsHardwareRequirements hardwareRequirements : resultset) {
```

Comment:
confirm to me: the flavor in the second position is better only than the flavor in the first position, right? I mean, have tested if the set is really ordered?

Reply:
Think about it, the simplest hardware requirements will always be on top of the stack.